src/renderer/contexts/GlobalNodeState.tsx

```diff
@@ -124,11 +124,11 @@ interface Global {
         inputId: InputId,
         inputSize: InputSize | undefined
     ) => readonly [Readonly<Size> | undefined, (size: Readonly<Size>) => void];
-    removeNodeById: (id: string) => void;
+    removeNodesById: (ids: string[]) => void;
```

Comment:
Whenever you declare a function takes a mutable array, I expect that function to modify the array in some way. However, that is not the case for pretty much all array parameters added by this PR. Could you please make them readonly?